omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java

```diff
@@ -92,6 +92,15 @@ public void shouldUpdateConceptAttributeType() throws Exception {
         Assert.assertEquals("org.openmrs.customdatatype.datatype.LongFreeTextDatatype", updatedConceptAttributeType.getDatatypeClassname());
     }
     
+    @Test
+   	public void shouldPurgeConceptAttributeType() throws Exception {
+       	Assert.assertNotNull(service.getConceptAttributeTypeByUuid(getUuid()));
+   		MockHttpServletRequest req = request(RequestMethod.DELETE, getURI() + "/" + getUuid());
+   		req.addParameter("purge", "true");
+   		handle(req);
+   		Assert.assertNull(service.getConceptAttributeTypeByUuid(getUuid()));
```

Comment:
@Irenyak1 have you taken a look at the logs are pointing out? https://github.com/openmrs/openmrs-module-webservices.rest/runs/4612493345?check_suite_focus=true#step:6:5125

Reply:
@kdaud  I checked the logs the moment it failed, I saw the errors that were being pointed out that's why I created a talk post here https://talk.openmrs.org/t/blocked-restws-875-end-point-concept-attribute-types-delete-test/35436. 
Did you have a chance to look at the error log in pastebin?

Reply:
As you can see here: https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-2.0/src/test/resources/conceptAttributeDataSet.xml
The uuid that you are trying to delete is referenced by a concept attribute and hence resulting into a referential integrity violation error in the logs. `Caused by: org.h2.jdbc.JdbcBatchUpdateException: Referential integrity constraint violation: "FK_6H9VJ62OUMIEKRGIUY885BS36: PUBLIC.CONCEPT_ATTRIBUTE FOREIGN KEY(ATTRIBUTE_TYPE_ID) REFERENCES PUBLIC.CONCEPT_ATTRIBUTE_TYPE(CONCEPT_ATTRIBUTE_TYPE_ID) (1)"; SQL statement:
delete from concept_attribute_type where concept_attribute_type_id=? [23503-187]`

So, just use a different uuid that is not referenced.

Reply:
> The uuid that you are trying to delete is referenced by a concept attribute and hence resulting into a referential integrity violation

That is very true, thanks @dkayiwa 

In fact, the referential Integrity constraint is enforced when a foreign key references the primary key of a relation.
So it specifies that all the values taken by the foreign key must either be available in the relation of the primary key or be null.

Reply:
Thanks @dkayiwa , thanks @kdaud

Reply:
@dkayiwa , @kdaud , @Muta-Jonathan , I have added to the reviews and made another commit. Could you please find time and review it. Thanks you very much.